base/dsfs/load.go

```diff
@@ -0,0 +1,302 @@
+		ds.Readme = rm
+	}
+	return nil
+}
+
+// loadReadme assumes the provided path is valid
+func loadReadme(ctx context.Context, fs qfs.Filesystem, path string) (st *dataset.Readme, err error) {
+	data, err := fileBytes(fs.Get(ctx, path))
+	if err != nil {
+		log.Debug(err.Error())
+		return nil, fmt.Errorf("error loading readme file: %w", err)
+	}
+	return dataset.UnmarshalReadme(data)
+}
+
+// LoadReadmeScript loads script data from a dataset path if the given dataset has a readme script is specified
+// the returned qfs.File will be the value of dataset.Readme.ScriptPath
+func LoadReadmeScript(ctx context.Context, fs qfs.Filesystem, dspath string) (qfs.File, error) {
+	ds, err := LoadDataset(ctx, fs, dspath)
+	if err != nil {
+		return nil, err
+	}
+
+	if ds.Readme == nil || ds.Readme.ScriptPath == "" {
+		return nil, ErrNoReadme
+	}
+
+	return fs.Get(ctx, ds.Readme.ScriptPath)
+}
+
+// DerefStats derferences a dataset's stats component if required
+// no-op if ds.Stats is nil or isn't a reference
+func DerefStats(ctx context.Context, store qfs.Filesystem, ds *dataset.Dataset) error {
+	if ds.Stats != nil && ds.Stats.IsEmpty() && ds.Stats.Path != "" {
+		sa, err := loadStats(ctx, store, ds.Stats.Path)
+		if err != nil {
+			log.Debug(err)
+			return fmt.Errorf("loading stats component: %w", err)
+		}
+		sa.Path = ds.Stats.Path
+		ds.Stats = sa
+	}
+	return nil
+}
+
+// loadStats assumes the provided path is valid
+func loadStats(ctx context.Context, fs qfs.Filesystem, path string) (sa *dataset.Stats, err error) {
+	data, err := fileBytes(fs.Get(ctx, path))
+	if err != nil {
+		log.Debug(err.Error())
+		return nil, fmt.Errorf("loading stats file: %w", err)
+	}
+	sa = &dataset.Stats{}
+	err = sa.UnmarshalJSON(data)
+	return sa, err
+}
+
+// DerefTransform derferences a dataset's transform element if required
+// should be a no-op if ds.Structure is nil or isn't a reference
+func DerefTransform(ctx context.Context, store qfs.Filesystem, ds *dataset.Dataset) error {
+	if ds.Transform != nil && ds.Transform.IsEmpty() && ds.Transform.Path != "" {
+		t, err := loadTransform(ctx, store, ds.Transform.Path)
+		if err != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("loading dataset transform: %w", err)
+		}
+		t.Path = ds.Transform.Path
+		ds.Transform = t
+	}
+	return nil
+}
+
+// loadTransform assumes the provided path is correct
+func loadTransform(ctx context.Context, fs qfs.Filesystem, path string) (q *dataset.Transform, err error) {
+	data, err := fileBytes(fs.Get(ctx, path))
+	if err != nil {
+		log.Debug(err.Error())
+		return nil, fmt.Errorf("error loading transform raw data: %s", err.Error())
+	}
+
+	return dataset.UnmarshalTransform(data)
+}
+
+// LoadTransformScript loads transform script data from a dataset path if the given dataset has a transform script specified
+// the returned qfs.File will be the value of dataset.Transform.ScriptPath
+// TODO - this is broken, assumes file is JSON. fix & test or depricate
```

Comment:
Is this comment still relevant or is it stale? Looks like it was added by me 2 years ago when dsfs moved into base from dataset.

Reply:
function is unused! I've removed it entirely